### PR TITLE
updates to mailchimp analysis for blog post

### PR DESCRIPTION
**Pull request by jthandy**

### CODE REVIEW

**Comment by jthandy** [comment]
@3mei I'd like to push this blog post live tomorrow. These changes aren't too big, lmk if you'll have trouble reviewing by then. Thanks!!

File: models/mailchimp/mailchimp_email_summary.sql [hunk 1]
```diff
@@ -35,11 +35,11 @@ unsubscribes as (
...
-    total_clicks, unsubscribed_date--,
-    --decode(bounced_date, null, 0, 1) as hard_bounced,
-    --decode(opened_date, null, 0, 1) as opened,
-    --decode(clicked_date, null, 0, 1) as clicked,
-    --decode(unsubscribed_date, null, 0, 1) as unsubscribed
+    total_clicks, unsubscribed_date,
+    decode(hard_bounced_date, null, 0, 1) as hard_bounced,
+    decode(first_opened_date, null, 0, 1) as opened,
+    decode(first_clicked_date, null, 0, 1) as clicked,
+    decode(unsubscribed_date, null, 0, 1) as unsubscribed
```
Comment (3mei) [inline models/mailchimp/mailchimp_email_summary.sql:42]: @jthandy Depending on the analysis you want to perform, sometimes you will want to look at the exact date and sometimes whether or not the date was recorded. The time series analysis looks only at the latter. Including both the timestamp and the decoded version here is redundant.

File: analysis/mailchimp/mailchimp_gains_losses.sql [hunk 2]
```diff
@@ -48,14 +48,25 @@ months as (
 	union
 	select month
 	from losses
+),
+
+monthly_gains_and_losses as (
+
+  select
+  	months.month, nvl(total_gains,0) as month_gains,
+  	nvl(total_losses,0) as month_losses
```
Comment (3mei) [inline analysis/mailchimp/mailchimp_gains_losses.sql]: @jthandy `-nlv(...)` Might as well just negate losses here to make everything more compact below.

File: analysis/mailchimp/mailchimp_gains_losses.sql [hunk 3]
```diff
@@ -48,14 +48,25 @@ months as (
...
-	months.month, nvl(total_gains,0) as total_gains,
-	nvl(total_losses,0) as total_losses
-from months
-left outer join gains
-	on months.month = gains.month
-left outer join losses
-	on months.month = losses.month
-order by months.month
\ No newline at end of file
+select month, month_gains, month_losses * -1 as month_losses, month_gains - month_losses as net_member_growth,
```
Comment (3mei) [inline analysis/mailchimp/mailchimp_gains_losses.sql]: @jthandy Can make this simpler:

`select month, month_gains, month_losses, month_gains + month_losses as net_member_growth,
	sum(month_gains) over (order by month rows unbounded preceding) as cum_gains,
	sum(month_losses) over (order by month rows unbounded preceding) as cum_losses,
	sum(month_gains+month_losses) over (order by month rows unbounded preceding) as active_members`

Comment (3mei) [inline analysis/mailchimp/mailchimp_gains_losses.sql]: @jthandy `select month, month_gains, month_losses, month_gains + month_losses as net_member_growth,
    sum(month_gains) over (order by month rows unbounded preceding) as cum_gains,
    sum(month_losses) over (order by month rows unbounded preceding) as cum_losses,
    sum(month_gains+month_losses) over (order by month rows unbounded preceding) as active_members`

**Comment by 3mei** [comment]
@jthandy Made a few comments. Note that there is a discrepancy between gists in the blogpost and this PR. Also, none of the additional analyses (top sends, most engaged subscribers) are included in the PR.

File: models/mailchimp/mailchimp_email_summary.sql (hunk 1 above)
Comment (jthandy) [inline models/mailchimp/mailchimp_email_summary.sql:42]: Yes but I used these in multiple subsequent analyses; I would prefer not to have to recalculate them repeatedly in every query. It's a really important fact about every email sent. And Redshift won't actually perform the aggregation if it's not needed to deliver the final results.

File: analysis/mailchimp/mailchimp_best_times.sql [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  select date_part(hr, sent_date), avg(opened::float)
+  from email_summary
+  group by 1
+  order by 1
+
+)
+
+select *
+from best_day_of_week
```
Comment (3mei) [inline analysis/mailchimp/mailchimp_best_times.sql:23]: @jthandy Looks good

File: analysis/mailchimp/mailchimp_most_engaged.sql [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  from {{env.schema}}.mailchimp_email_summary
+
+)
+
+select email_id, sum(clicked)::float / sum(opened)
+  from email_summary
+ group by 1
+having sum(opened) > 5
+ order by 2 desc
+ limit 100
```
Comment (3mei) [inline analysis/mailchimp/mailchimp_most_engaged.sql]: @jthandy Looks good as well, just spacing issues.

**Pull request merged**